src/openforms/pre_requests/tests/test_clients.py

```diff
@@ -0,0 +1,47 @@
+from unittest.mock import patch
+
+from django.test import TestCase
+
+import requests_mock
+
+from openforms.prefill.contrib.haalcentraal.tests.utils import load_binary_mock
+from openforms.registrations.contrib.zgw_apis.tests.factories import ServiceFactory
+
+from ..base import PreRequestHookBase
+from ..registry import Registry
+
+
+class PreRequestHooksTest(TestCase):
+    def test_pre_request_hook(self):
+        register = Registry()
+
+        @register("test-hook")
+        class PreRequestHook(PreRequestHookBase):
+            def __call__(self, url, method, **kwargs):
+                headers = kwargs.get("headers", {})
+                headers.update({"test": "test"})
```

Comment:
A more realistic example would ensure the headers are set:

```suggestion
            def __call__(self, url, method, **kwargs):
                kwargs.setdefault("headers", {})
                kwargs["headers"].update({"test": "test"})
```